main.tf

```diff
@@ -0,0 +1,84 @@
+data "aws_availability_zones" "available" {}
+
+data "aws_vpc" "default" {
+  id = "${var.vpc_id}"
+}
+
+module "subnets_label" {
+  source     = "git::https://github.com/cloudposse/terraform-null-label.git?ref=tags/0.2.2"
+  namespace  = "${var.namespace}"
+  name       = "${var.name}"
+  stage      = "${var.stage}"
+  delimiter  = "${var.delimiter}"
+  attributes = ["subnet"]
+  tags       = "${var.tags}"
+}
+
+resource "aws_subnet" "default" {
+  count             = "${length(compact(var.names)) * length(compact(var.availability_zones))}"
+  vpc_id            = "${var.vpc_id}"
+  availability_zone = "${element(var.availability_zones, count.index)}"
+  cidr_block        = "${cidrsubnet(length(var.base_cidr) > 0 ? var.base_cidr : data.aws_vpc.default.cidr_block, ceil(log(length(compact(var.names)) * length(compact(var.availability_zones)) * 2, 2)), count.index)}"
+
+  tags = {
+    "Name"      = "${module.subnets_label.id}${var.delimiter}${element(var.names, count.index)}${var.delimiter}${element(var.availability_zones, count.index)}"
+    "Stage"     = "${module.subnets_label.stage}"
+    "Namespace" = "${module.subnets_label.namespace}"
+  }
+}
+
+## NGW
+resource "aws_eip" "default" {
+  count = "${var.nat_enabled ? length(compact(var.names)) * length(compact(var.availability_zones)) : 0}"
```

Comment:
Use a `local` variable for `length(compact(var.names)) * length(compact(var.availability_zones)`